templates/registry-standalone/overlay/opt/corral/registry/registry-install.sh

```diff
@@ -133,11 +137,48 @@ else
 	if [ "enabled" = "$CORRAL_windows_registry" ]; then
 		corral_log "Adding Windows images to registry"
 		wget -O rancher-windows-images.txt "${DOWNLOAD_URL}v${CORRAL_rancher_version}"/rancher-windows-images.txt
+
+		source /etc/os-release
+
+		if [[ $NAME == *"SUSE"* || $NAME == *"SLES"* ]]; then
+			sudo zypper addrepo --refresh https://download.opensuse.org/repositories/system:/snappy/openSUSE_Leap_$VERSION_ID snappy
+			sudo zypper --gpg-auto-import-keys refresh
+			sudo zypper dup --from snappy
+			sudo zypper install snapd
+			source /etc/profile
+			sudo systemctl enable --now snapd
+			sudo systemctl enable --now snapd.apparmor
+			sudo snap install snap-store
+
+		elif [[ $NAME == *"Red Hat"* ]]; then
+			if [[ $VERSION_ID == *"7."* ]]; then
+				sudo rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+			else
+				sudo dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-${VERSION_ID:0:1}.noarch.rpm
+				sudo dnf -y upgrade
+			fi
+			sudo subscription-manager repos --enable "rhel-*-optional-rpms" --enable "rhel-*-extras-rpms"
+			sudo yum -y update
+			sudo yum -y install snapd
+			sudo systemctl enable --now snapd.socket
+			sudo ln -s /var/lib/snapd/snap /snap
+
+
+			sleep 10
```

Comment:
Not against the sleep as we do it after install cert manager. But is the reason?

Reply:
for some reason, RHEL took some time to finish the install of snap after that last line ran. I assumed it was an OS specific thing, but when I was testing it would complain that the snap store wasn't up yet. After I put this sleep, it worked every time though